runtime/runtime-params-estimator/src/testbed_runners/qemu.rs

```diff
@@ -0,0 +1,108 @@
+//! QEMU instrumentation code to get used resources as measured by the QEMU plugin.
+
+use std::{ops, os::raw::c_void};
+
+use num_rational::Ratio;
+
+// We use several "magical" file descriptors to interact with the plugin in QEMU
+// intercepting read syscall. Plugin counts instructions executed and amount of data transferred
+// by IO operations. We "normalize" all those costs into instruction count.
+const CATCH_BASE: u32 = 0xcafebabe;
+const HYPERCALL_START_COUNTING: u32 = 0;
+const HYPERCALL_STOP_AND_GET_INSTRUCTIONS_EXECUTED: u32 = 1;
+const HYPERCALL_GET_BYTES_READ: u32 = 2;
+const HYPERCALL_GET_BYTES_WRITTEN: u32 = 3;
+
+#[derive(Clone, PartialEq, Eq)]
+pub(crate) struct QemuMeasurement {
+    // Integers are not enough because results are averaged over transactions in a block.
+    pub instructions: Ratio<u64>,
```

Comment:
Let's use use just `u64` here -- it seems that this layer of abstraction should return just a raw number for single timed interval, while the logic for averaging that out over several measurements should live above it.